backend/internal/storage/postgres/assignment_outlines.go

```diff
@@ -84,62 +86,60 @@ func (db *DB) GetAssignmentByID(ctx context.Context, assignmentID int64) (models
 	return assignmentOutline, nil
 }
 
-func (db *DB) GetAssignmentWithTemplateByAssignmentID(ctx context.Context, assignmentID int64) (models.AssignmentOutlineWithTemplate, error) {
-	var result models.AssignmentOutlineWithTemplate
-	err := db.connPool.QueryRow(ctx, `
-	SELECT 
-		ao.id, ao.template_id, ao.created_at, ao.released_at, ao.name, ao.classroom_id, ao.rubric_id, ao.group_assignment, ao.main_due_date,
-		at.template_repo_owner, at.template_repo_id, at.template_repo_name, at.created_at
-	FROM assignment_outlines ao
-	JOIN assignment_templates at ON ao.template_id = at.template_repo_id
-	WHERE ao.id = $1`, assignmentID).Scan(
-		&result.ID,
-		&result.TemplateID,
-		&result.CreatedAt,
-		&result.ReleasedAt,
-		&result.Name,
-		&result.ClassroomID,
-		&result.RubricID,
-		&result.GroupAssignment,
-		&result.MainDueDate,
-		&result.Template.TemplateRepoOwner,
-		&result.Template.TemplateID,
-		&result.Template.TemplateRepoName,
-		&result.Template.CreatedAt,
-	)
-	if err != nil {
-		return models.AssignmentOutlineWithTemplate{}, errs.NewDBError(err)
-	}
-
-	return result, nil
-}
-
-func (db *DB) CreateAssignment(ctx context.Context, assignmentRequestData models.AssignmentOutline) (models.AssignmentOutline, error) {
-	var assignmentData models.AssignmentOutline
+func (db *DB) CreateAssignment(ctx context.Context, assignmentRequestData models.AssignmentOutline) (int32, error) {
```

Comment:
do we really need to have an int32 here

Reply:
I am not sure what you mean. The `AssignmentOutline.id` field is of type `int32`. Thus, to return the ID, it's necessary to have that. `int` != `int32` in Go.

Reply:
this is covered by the "make all IDs the same type of int" ticket. Fine to keep for now